appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/processor/ApplicationProcessor.java

```diff
@@ -760,12 +756,63 @@ public void visitParameter(Parameter parameter, AnnotatedElement element, ApiCon
                         mediaType.setSchema(new SchemaImpl());
                     }
                     mediaType.getSchema()
-                            .setType(ModelUtils.mergeProperty(mediaType.getSchema().getType(), type, false));
+                    .setType(ModelUtils.mergeProperty(mediaType.getSchema().getType(), type, false));
                 }
             }
         }
     }
 
+    private static org.eclipse.microprofile.openapi.models.parameters.Parameter findOperationParameterFor(Parameter parameter,
+            Method annotated, ApiContext context) {
+        String name = parameter.name();
+        // If the parameter reference is valid
+        if (name != null && !name.isEmpty()) {
+            // Get all parameters with the same name
+            List<java.lang.reflect.Parameter> matchingMethodParameters = Arrays
+                    .asList(annotated.getParameters()).stream()
+                    .filter(x -> name.equals(ModelUtils.getParameterName(x)))
+                    .collect(Collectors.toList());
+            // If there is more than one match, filter it further
+            ParameterIn in = parameter.in();
+            if (matchingMethodParameters.size() > 1 && in != null && in != ParameterIn.DEFAULT) {
+                // Remove all parameters of the wrong input type
+                matchingMethodParameters
+                .removeIf(x -> ModelUtils.getParameterType(x) != In.valueOf(in.name()));
+            }
+            if (matchingMethodParameters.isEmpty()) {
+                return null;
```

Comment:
FYI: This is a fix I added. 1.0 TCK simply had no case that would otherwise fail the next line where we always get the first entry of the list. But some cases do not have a matching parameter.